db/fetcher/fetcher.go

```diff
@@ -636,21 +681,11 @@ func (df *DocumentFetcher) fetchNext(ctx context.Context) (EncodedDocument, Exec
 			} else {
 				return df.doc, df.execInfo, nil
 			}
+		}
 
-			if !spansDone {
-				continue
-			}
-
+		if spansDone {
```

Comment:
additional placeholder comment while I review the flow/logic of these changes. (see above comment)

Reply:
All good, can resolve!